apps/api/src/server/applications/examination-timetable-items/examination-timetable-items.controller.js

```diff
@@ -33,5 +35,34 @@ export const createExaminationTimetableItem = async (_request, response) => {
 	const { body } = _request;
 	const examinationTimetableItem = await exminationTimetableItemsRepository.create(body);
 
+	// find the examination folder id to create subfolder.
+	let examinationFolder = await folderRepository.getFolderByNameAndCaseId(
+		body.caseId,
+		'Examination'
+	);
+
+	if (!examinationFolder) {
+		examinationFolder = await folderRepository.createFolder({
+			displayNameEn: 'Examination',
+			caseId: body.caseId,
+			parentFolderId: null,
+			displayOrder: 100
+		});
+	}
+
+	// in some case we don't have date, use mandatory field end date in that case.
+	const date = examinationTimetableItem.date || examinationTimetableItem.endDate;
```

Comment:
I'm maybe misunderstanding but 'date' and 'endDate' should be the same thing - examination items will always have a 'date' which is the 'endDate', but deadline items will also have a 'startDate' (in which case, the 'date' is the 'endDate').

So this logic should just always be taking the 'date' which is mandatory. I'm wondering if the entity has been misconfigured?